zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/archiver/ProcessInstanceToBeArchivedCountJob.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.tasks.archiver;
+
+import co.elastic.clients.elasticsearch.ElasticsearchAsyncClient;
+import io.camunda.exporter.metrics.CamundaExporterMetrics;
+import io.camunda.exporter.tasks.BackgroundTask;
+import io.camunda.exporter.tasks.util.ElasticsearchArchiverQueries;
+import java.util.concurrent.CompletionStage;
+
+public class ProcessInstanceToBeArchivedCountJob implements BackgroundTask {
+  public static final int DELAY_BETWEEN_RUNS = 60000;
+  public static final int MAX_DELAY_BETWEEN_RUNS = 300000;
+
+  private final CamundaExporterMetrics metrics;
+  private final ElasticsearchAsyncClient client;
+  private final int partitionId;
+  private final String archivingTimePoint;
+  private final String listViewIndexName;
+
+  public ProcessInstanceToBeArchivedCountJob(
+      final CamundaExporterMetrics metrics,
+      final ElasticsearchAsyncClient client,
+      final int partitionId,
+      final String archivingTimePoint,
+      final String listViewIndexName) {
+    this.metrics = metrics;
+    this.client = client;
+    this.partitionId = partitionId;
+    this.archivingTimePoint = archivingTimePoint;
+    this.listViewIndexName = listViewIndexName;
+  }
+
+  @Override
+  public CompletionStage<Integer> execute() {
+    final var countRequest =
+        ElasticsearchArchiverQueries.createFinishedInstancesCountRequest(
+            listViewIndexName, archivingTimePoint, partitionId);
+    final var countFuture = client.count(countRequest);
+
+    return countFuture
+        .whenCompleteAsync(
+            (res, err) -> {
+              if (err == null) {
```

Comment:
🔧 I would add a trace log on failure, so we have some escape hatch to debug if it fails (e.g. "why don't I get this metric?"). At trace level, I think it's fine to have it - anything else we typically wouldn't care about failures.